src/Data/Aeson/Commit.hs

```diff
@@ -8,19 +8,22 @@ import           Data.Aeson.Types
 import           Data.Text            (Text)
 import           Data.Void            (Void, absurd)
 
--- | A parser that has _two_ failure modes; the 'ExceptT' or in the underlying 'Parser'.
---   The alternative instance only recovers from failures in the `ExceptT`.
---   This means that, as soon as we successfully construct a 'Right' value, the 'Alternative' considers the 'Commit' a success, even though the inner 'Parser' can still fail.
+-- | A 'Parser' that has _two_ failure modes; recoverable and non-recoverable.
+--   The default, recoverable failure is the equivalent to aeson's default 'Parser' behavior.
+--   The non-recoverable failure mode is used to commit to a branch; to commit means that every subsequent failure is non-recoverable.
 --
---   The 'Void' guarantees that that parser contains an error value.
+--   The implementation works by capturing failure in either the 'ExceptT' or in the underlying 'Parser'.
+--   The derived 'Alternative' instance will only recover from failures in the 'ExceptT'.
```

Comment:
I think the logic is reversed, we don't recover from the failures in the `ExceptT` right?

Reply:
Yes we do? A `Right <parser failure>` counts as success to `Alternative`, so it won't recover from it.